microsoft-edge/webview2/get-started/hololens2.md

```diff
@@ -0,0 +1,627 @@
+
+
+<!-- ====================================================================== -->
+## Learn about WebView2 events and interactions in Unity
+
+The Microsoft Mixed Reality WebView plugin for Unity provides events for working with the webview. 
+
+One important event is `IWithPostMessage.MessageReceived`, which is raised when a message is sent from the webview to the Unity app. The `MessageReceived` event is defined in the `Microsoft.MixedReality.WebView.IWithPostMessage` interface. 
+
+`IWithPostMessage` also defines the `PostMessage()` method, which can be used to send a message from the Unity app to the webview.
+
+Here is an example of how to use these events and methods for working with the webview:
+
+```C#
+using Microsoft.MixedReality.WebView;
+
+public class WebViewExample : MonoBehaviour, IWithPostMessage
+{
+    private void Start()
+    {
+        var webViewComponent = gameObject.GetComponent<WebView>();
+
+        if (Url != string.Empty)
+        {
+            Debug.Log("Loading URL: " + Url);
+            webViewComponent.Load(new Uri(Url));
+        }
+
+        webViewComponent.GetWebViewWhenReady((IWebView webView) =>
+        {
+            // Add event listeners for WebView2 events
+            ((IWithPostMessage)webView).MessageReceived += OnMessageReceived;
+        });
+    }
+
+    // Handler for WebView2 OnPostMessage event
+    void OnMessageReceived(string message)
+    {
+        Debug.Log("WebView2 message received: " + message);
+    }
+}
+```
+
+<!-- ====================================================================== -->
+## Additional considerations for WebView2 on HoloLens 2
+
+
+<!-- ------------------------------ -->
+#### Limitations and known issues
+
+When developing a HoloLens 2 Unity app with WebView2, be aware of some limitations and known issues:
+
+* **Pop-ups**: Pop-ups don't work well within WebView2 inside Unity apps on HoloLens 2, but they work fine in 2D XAML apps on the device. Avoid pop-ups and use alternative techniques or UI designs, such as custom pop-up-like elements within the WebView using HTML, CSS, and JavaScript.
+
+* **New windows**: WebView2 instances on HoloLens 2 navigate within the same window by default, unlike on Desktop. Follow this default behavior for a better user experience. Additionally, the DevTools window cannot be launched.
+
+* **Enterprise authentication**: Automatic Single Sign-On (SSO) leveraging OS-level tokens is currently not supported in WebView2 on HoloLens 2. Users can still sign in by providing credentials, except for cases requiring device-level authentication. Cookie storage works as expected.
+
+* **User interactions**: Unlike native HoloLens 2 slates, WebView2 is best interacted with using far-interaction hand rays. Touch-to-swipe and scroll interactions might not be supported.
```

Comment:
```suggestion
* **User interactions**: Unlike native HoloLens 2 slates, WebView2 is best interacted with by using far-interaction hand rays. Touch-to-swipe and scroll interactions might not be supported.
```
d/k phrase "far-interaction hand rays"

https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/u/using-by-using-with -- "In instructions or technical content, use _by using_ or _that use_ to eliminate ambiguity."

Reply:
added "by" locally

Reply:
latest Files Changed tab has "by"
https://github.com/MicrosoftDocs/edge-developer/pull/2567/files

can close this thread